.github/workflows/coccinelle.yaml

```diff
@@ -0,0 +1,28 @@
+name: Coccinelle
+on:
+  pull_request:
+  push:
+jobs:
+  memory_leak:
+    name: Coccinelle
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Install Dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install software-properties-common
+        sudo add-apt-repository ppa:npalix/coccinelle
+        sudo apt-get update
+        sudo apt-get install coccinelle
+
+    - name: Checkout TimescaleDB
+      uses: actions/checkout@v2
+
+    - name: Run coccinelle
+      run: |
+        for f in coccinelle/*.cocci; do
+          find . -name '*.c' | xargs spatch --very-quiet -sp_file $f | tee -a coccinelle.diff
+        done
+
+        if [ -s coccinelle.diff ] ; then exit 1; fi
```

Comment:
Is this diff uploaded somewhere?

Reply:
I see that it goes to stderr, OK https://github.com/timescale/timescaledb/runs/4024942886?check_suite_focus=true